### PR TITLE
check actions to process in the interceptor

### PR DESCRIPTION
**Pull request by akihikokuroda**
# Changes
This PR is for https://github.com/tektoncd/experimental/issues/250

The pull request is processed only when the action is one of opened, reopened or synchronize.

The event listener sends the actions to be processed in the header to the interceptor.  The interceptor checks if the action in the event matches one of the actions sent by the event listener if the actions are not empty.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: webhooks-extension/cmd/interceptor/interceptor.go [hunk 1]
```diff
@@ -103,8 +104,19 @@ func main() {
...
+					wantedActions := request.Header["Wext-Incoming-Actions"]
+					if len(wantedActions) == 0 {  
+						validationPassed = true
+						log.Printf("[%s] Validation PASS (repository URL, secret payload, event type checked)", foundTriggerName)
+					} else {
+						actions := strings.Split(wantedActions[0], ",")
+						for _, action := range actions {
+							if action == result.Action {
+								validationPassed = true
+								log.Printf("[%s] Validation PASS (repository URL, secret payload, event type, action checked)", foundTriggerName)
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply webhooks-extension/cmd/interceptor/interceptor.go]: @ncskier Thanks for review.  I updated it.

File: webhooks-extension/cmd/interceptor/interceptor.go [hunk 2]
```diff
@@ -113,8 +114,19 @@ func main() {
...
+						validationPassed = true
+						log.Printf("[%s] Validation PASS (repository URL, secret payload, event type checked)", foundTriggerName)
+					} else {
+						actions := strings.Split(wantedActions[0], ",")
+						for _, action := range actions {
+							if action == result.Action {
+								validationPassed = true
+								log.Printf("[%s] Validation PASS (repository URL, secret payload, event type, action:%s checked)", foundTriggerName, action)
+							}
+						}
```
Comment (ncskier) [inline webhooks-extension/cmd/interceptor/interceptor.go]: I think that this change surfaces a bug in the existing interceptor code. When `validationPassed` is `false`, it is never written as an error. On line 155 (https://github.com/tektoncd/experimental/pull/354/files#diff-4e40b3eebcbb16692d761b108f341566R155) we should have an `else` statement to catch the case when `validationPassed` is false. For example:
```go
if validationPassed {
	log.Printf("[%s] Validation PASS so writing response", foundTriggerName)
	_, err := writer.Write(payload)
	if err != nil {
		log.Printf("[%s] Failed to write response for Github event ID: %s. Error: %s", foundTriggerName, id, err.Error())
		http.Error(writer, fmt.Sprint(err), http.StatusInternalServerError)
		return
	}
} else {
	http.Error(writer, "Validation failed", http.StatusExpectationFailed)
}
```

Comment (akihikokuroda) [reply webhooks-extension/cmd/interceptor/interceptor.go]: @ncskier Good catch!!  I fix it.  Thanks!

**Pull request merged**